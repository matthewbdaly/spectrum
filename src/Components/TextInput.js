// @flow
import React, { Component } from 'react';
import type { Node } from 'react';
import './TextInput.scss';

type Props = {
  children?: Node
};

class TextInput extends Component<Props> {
  render() {
    return (
      <label className="textinput">{this.props.children}
        <input type="text"/>
      </label>
    );
  }
}

export default TextInput;
