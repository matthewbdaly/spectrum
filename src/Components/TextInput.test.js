import React from 'react';
import ReactDOM from 'react-dom';
import TextInput from './TextInput';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextInput />, div);
});

it('should match the snapshot', () => {
  const div = document.createElement('div');
  const el = ReactDOM.render(<TextInput />, div);
  expect(el).toMatchSnapshot();
});
