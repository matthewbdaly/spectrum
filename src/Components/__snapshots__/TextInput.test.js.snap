// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
TextInput {
  "_reactInternalFiber": FiberNode {
    "_debugID": 9,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "<PROJECT_ROOT>/src/Components/TextInput.test.js",
      "lineNumber": 12,
    },
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugID": 10,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "<PROJECT_ROOT>/src/Components/TextInput.js",
        "lineNumber": 13,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugID": 11,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": Object {
          "fileName": "<PROJECT_ROOT>/src/Components/TextInput.js",
          "lineNumber": 14,
        },
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childExpirationTime": 0,
        "contextDependencies": null,
        "effectTag": 0,
        "elementType": "input",
        "expirationTime": 0,
        "firstEffect": null,
        "index": 1,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "type": "text",
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "type": "text",
        },
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": <input
          type="text"
/>,
        "tag": 5,
        "treeBaseDuration": 0,
        "type": "input",
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "contextDependencies": null,
      "effectTag": 0,
      "elementType": "label",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          undefined,
          <input
            type="text"
/>,
        ],
        "className": "textinput",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": Array [
          undefined,
          <input
            type="text"
/>,
        ],
        "className": "textinput",
      },
      "ref": null,
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": <label
        class="textinput"
>
        <input
                type="text"
        />
</label>,
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "label",
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "contextDependencies": null,
    "effectTag": 1,
    "elementType": [Function],
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {},
    "memoizedState": null,
    "mode": 0,
    "nextEffect": FiberNode {
      "_debugID": 7,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": FiberNode {
        "_debugID": 7,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": [Circular],
        "child": null,
        "childExpirationTime": 0,
        "contextDependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 1073741823,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <label
                        class="textinput"
            >
                        <input
                                    type="text"
                        />
            </label>
</div>,
          "context": Object {},
          "current": [Circular],
          "didError": false,
          "earliestPendingTime": 0,
          "earliestSuspendedTime": 0,
          "expirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "interactionThreadID": 2,
          "latestPendingTime": 0,
          "latestPingedTime": 0,
          "latestSuspendedTime": 0,
          "memoizedInteractions": Set {},
          "nextExpirationTimeToWorkOn": 0,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <TextInput />,
            },
            "tag": 0,
          },
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <TextInput />,
            },
            "tag": 0,
          },
        },
      },
      "child": [Circular],
      "childExpirationTime": 0,
      "contextDependencies": null,
      "effectTag": 32,
      "elementType": null,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <TextInput />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <label
                    class="textinput"
          >
                    <input
                              type="text"
                    />
          </label>
</div>,
        "context": Object {},
        "current": [Circular],
        "didError": false,
        "earliestPendingTime": 0,
        "earliestSuspendedTime": 0,
        "expirationTime": 0,
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "interactionThreadID": 2,
        "latestPendingTime": 0,
        "latestPingedTime": 0,
        "latestSuspendedTime": 0,
        "memoizedInteractions": Set {},
        "nextExpirationTimeToWorkOn": 0,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "pendingInteractionMap": Map {},
        "pingCache": null,
        "timeoutHandle": -1,
      },
      "tag": 3,
      "treeBaseDuration": 0,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <TextInput />,
        },
        "firstCapturedEffect": null,
        "firstCapturedUpdate": null,
        "firstEffect": null,
        "firstUpdate": null,
        "lastCapturedEffect": null,
        "lastCapturedUpdate": null,
        "lastEffect": null,
        "lastUpdate": null,
      },
    },
    "pendingProps": Object {},
    "ref": null,
    "return": FiberNode {
      "_debugID": 7,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": FiberNode {
        "_debugID": 7,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": [Circular],
        "child": null,
        "childExpirationTime": 0,
        "contextDependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 1073741823,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <label
                        class="textinput"
            >
                        <input
                                    type="text"
                        />
            </label>
</div>,
          "context": Object {},
          "current": [Circular],
          "didError": false,
          "earliestPendingTime": 0,
          "earliestSuspendedTime": 0,
          "expirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "interactionThreadID": 2,
          "latestPendingTime": 0,
          "latestPingedTime": 0,
          "latestSuspendedTime": 0,
          "memoizedInteractions": Set {},
          "nextExpirationTimeToWorkOn": 0,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <TextInput />,
            },
            "tag": 0,
          },
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <TextInput />,
            },
            "tag": 0,
          },
        },
      },
      "child": [Circular],
      "childExpirationTime": 0,
      "contextDependencies": null,
      "effectTag": 32,
      "elementType": null,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <TextInput />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <label
                    class="textinput"
          >
                    <input
                              type="text"
                    />
          </label>
</div>,
        "context": Object {},
        "current": [Circular],
        "didError": false,
        "earliestPendingTime": 0,
        "earliestSuspendedTime": 0,
        "expirationTime": 0,
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "interactionThreadID": 2,
        "latestPendingTime": 0,
        "latestPingedTime": 0,
        "latestSuspendedTime": 0,
        "memoizedInteractions": Set {},
        "nextExpirationTimeToWorkOn": 0,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "pendingInteractionMap": Map {},
        "pingCache": null,
        "timeoutHandle": -1,
      },
      "tag": 3,
      "treeBaseDuration": 0,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <TextInput />,
        },
        "firstCapturedEffect": null,
        "firstCapturedUpdate": null,
        "firstEffect": null,
        "firstUpdate": null,
        "lastCapturedEffect": null,
        "lastCapturedUpdate": null,
        "lastEffect": null,
        "lastUpdate": null,
      },
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": [Circular],
    "tag": 1,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
